### DISCUSSION TITLE
Debian package, installation instructions

### DISCUSSION THREAD

**Discussion by asarhaddon** [NanoVNA-Saver/nanovna-saver#626] [General]
Hello.

I have prepared a Debian package, which should appear in
- [x] Debian experimental               (within days?)
- [ ] Debian testing (via unstable)   (within months?)
- [ ] Ubuntu                                     (within months?)
- [ ] Debian stable                           (within years?)

Provided appropriate repository settings, anyone interested should then be able to try it and find more details at
https://tracker.debian.org/pkg/nanovna-saver.

I suggest to take the opportunity and split the installation instructions.
`README.rst` should tell end users without programming to install Windows executables, or integrated `.deb` packages or similar.
Documentation for contributors should give instructions about cloning, virtual environments and so on.

Once the human-written, integrated `.deb` is available, I think that `make deb` will be obsolete.

Here is a draft suggestion.

### Normal user

A `.deb` package tailored for your distribution should be available from your usual repositories.
It is recommended unless you intend to modify the source code.

    [sudo] apt install nanonva-saver    # Add sudo on Ubuntu derivatives.

### Advanced user needing small local changes

If you need to change the source code for local purposes, you may build a `.deb` package then install it cleanly as above.
The `dch` command should suggest a version like 0.6.0-3.1 if the currently
available version was 0.6.0-1 (0.6.0:upstream, 3:Debian 1:local).

    git clone https://salsa.debian.org/debian/nanovna-saver.git
    cd nanonva-saver
    [sudo] apt-get build-dep
    # apply changes
    dch
    dpkg-buildpackage
    [sudo] apt install ../nanonva-saver_*.deb

### Contributor installing their changes

If you intend to eventually forward your changes to the upstream authors, and if you are at ease with potential git merge conflicts, you should ideally work on latest upstream sources.
Insert the following lines between `apt-get build-dep` and `# apply changes`.

    git remote add github https://github.com/NanoVNA-Saver/nanovna-saver
    git fetch github
    git merge remotes/github/main

### Contributor not installing their changes

(current `Ubuntu 20.04 / 22.04` section)

**Comment by Ho-Ro** [comment]
I will check this when I'm back on my system - thx.
My `make deb` was the quick'n'dirty way how I install my local changes with the option of clean removal (on my system), it was never intended for distribution. (More worse - I have a lot of devel projects on my machine where I use `checkinstall` for this task, it's still better than `make install`).
